cql3/restrictions/statement_restrictions.cc

```diff
@@ -289,7 +285,12 @@ static std::vector<expr::expression> extract_clustering_prefix_restrictions(
             current_binary_operator = nullptr;
         }
 
-        void operator()(const column_value_tuple&) {
+        void operator()(const tuple_constructor& tc) {
+            for (auto& e : tc.elements) {
+                if (!std::holds_alternative<column_value>(e)) {
```

Comment:
Why do a dynamic check here but not in evaluate()?

Reply:
This is in the prepare phase, where we're very sensitive to shapes. For example, if we relax the the grammar to allow `WHERE (?, ?) = pk`, this area will need adjustment while evaluate() will just work (well, after we change term to expression). evaluate() is not sensitive to patterns.

(somewhat handwavy but I think correct)

Reply:
Handwave accepted. ;)